iothub/service/src/AmqpServiceClient.cs

```diff
@@ -88,6 +93,11 @@ public async override Task CloseAsync()
 
         public async override Task SendAsync(string deviceId, Message message, TimeSpan? timeout = null)
         {
+            if (Logging.IsEnabled)
+            {
+                Logging.Enter(this, $"Sending message with Id [{message.MessageId}] for device [{deviceId}]", nameof(SendAsync));
```

Comment:
What if message is null?

Reply:
good point, I'll check if we do allow message to be null, and if yes, then update this

Reply:
Interestingly message can be null, i'll update the log statement

Reply:
Logging should never cause an exception, so we have to be sure it is always safe.

Reply:
yes, I agree 100%.